components/org.wso2.carbon.identity.organization.management.service/src/main/java/org/wso2/carbon/identity/organization/management/service/dao/impl/CacheBackedOrganizationManagementDAO.java

```diff
@@ -155,9 +162,8 @@ public void deleteOrganization(String organizationId) throws OrganizationManagem
 
         String tenantDomain = resolveTenantDomain(organizationId);
         organizationMgtDAO.deleteOrganization(organizationId);
-        if (tenantDomain != null) {
-            clearOrganizationDetailsCache(organizationId, tenantDomain);
-        }
+        clearOrganizationCache(organizationId, tenantDomain);
+        clearBasicOrganizationCacheOfParent(organizationId);
```

Comment:
Are we keeping the org cache for parent ? what do we achieve with it

Reply:
https://github.com/wso2/identity-organization-management-core/pull/194/commits/89824b49512ce1bbde2edc2412f0f790faa0df69